extra/import_data.py

```diff
@@ -1,197 +1,158 @@
-
-                            # add variant info along with metadata to collection
-                            variant.update(patient)
-                            db.insert_one(variant)
-            bar.update(bar.value)
-        logger.debug('finished uploading: %s' % file)
-        # Moves file to other folder, to make uploading a new file (with same filename) possible on the UMCU server (for replacing)
-        os.replace(file, os.path.dirname(__file__) + "/verwerkt/" + os.path.basename(file))
-    print("## \t\tFinished in {0:.2f} minutes".format((time.time() - start_time) / 60))
-
-
-main()
+    '''Main function, for extracting the data from the Alissa output and send to function to parse to MongoDB'''
+     # connection with MongoDB and the correct database "vus" and collection "lastUpdatedOn"
+    mongoclient = pymongo.MongoClient("mongodb://localhost:27017/")
+    db = mongoclient.vus.lastUpdatedOn
+    
+    # Retrieve the last time MongoDB has updated itself, to filter the latest Alissa analyses only
+    lastUpdatedOn_mongoDB = db.find_one({"version" : 1}, {"lastUpdatedOn":1, "_id":0})
+    #If not exist yet, use '2018-01-01T00:00:00.000+0000' and add to mongo (first time use)
+    if lastUpdatedOn_mongoDB is None:
+        db.replace_one({"version" : 1}, {"lastUpdatedOn" : '2018-01-01T00:00:00.000+0000', "version" : 1}, True)
+        lastUpdatedOn_mongoDB = db.find_one({"version" : 1}, {"lastUpdatedOn":1, "_id":0})
+    
+    for key, value in lastUpdatedOn_mongoDB.items():
+        lastUpdatedOn_mongoDB = value
+    
+    # retrieving data from Alissa
+    analysis = None
+    for analysis in client.get_analyses(status='COMPLETED', 
+                                        lastUpdatedAfter=lastUpdatedOn_mongoDB, # format is '2020-01-01T00:00:00.000+0000'
+                                        analysisPipelineName='ONB01',
+                                        analysisType='INHERITANCE'): #add for testing [:x], where x is numer of iterations
+        if not analysis['classificationTreeName']:  # Skip analysis
+            continue
+        else:
+            # retrieve basic info from Alissa about the analysis
+            patient_dn_no = client.get_report(analysis['id'])[0]['reportName'].split("_")[0]
+            logger.info('start Alissa retrieval of: %s' % patient_dn_no)
+            ONB01_analysis = client.get_inheritance_analyses(analysis['id'])
+            accession_number = client.get_patient(analysis['patientId'])['accessionNumber']
+            export_id = client.post_inheritance_analyses_variants_export(analysis['id'], marked_review=True, marked_include_report=False)['exportId']
+            analyis_sources = client.get_sources(analysis['id'])
+            
+            # retrieve the VUS/GUS info based on the analysis ID
+            VUS_response = None
+            while VUS_response is None:
+                try:
+                    time.sleep(5)  # 5 sec delay to request exported report.
```

Comment:
How often do exports take longer than 5 seconds? In `manual_review_analyses.py` I have based the sleep time on number of variants.

Reply:
my number of variants is mostly only 2, and sometimes 8 or higher. It doesn't often need more than 5 seconds, except if the Alissa database is slow. With testing I saw this usually 2-3 times for 10 analysis, but also many times where there were none during 20+ analysis. I think for such small numbers it feels not really necessary to base it on that instead just a number. Maybe you could give me an advise on this?

Reply:
Sounds fine to use 5 seconds in this case. I export sometimes 1000+ variants were it can take much longer.

Reply:
👍